### PR TITLE
feat(CompactSizeUint): encode/decode writer

### PR DESCRIPTION
**Pull request by tdelabro**

### CODE REVIEW

File: src/types/CompacSizeUint.zig [hunk 1]
```diff
@@ -102,45 +117,88 @@ pub fn decode(input: []const u8) DecodeSelfError!Self {
...
+
     return .{ .inner = @bitCast(buffer) };
 }
 
 // TESTS
 
 test "ok_full_flow_for_key_values" {
     const values = [_]u64{ 0, 252, 0xffff, 0xffffffff, std.math.maxInt(u64) };
+    const zeroed_buffer = [9]u8{ 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+    var buffer = [9]u8{ 0, 0, 0, 0, 0, 0, 0, 0, 0 };
```
Comment (tcoratger) [inline src/types/CompacSizeUint.zig:161]:
```suggestion
    const zeroed_buffer = [_]u8{0} ** 9;
    var buffer = [_]u8{0} ** 9;
```

File: src/types/CompacSizeUint.zig [hunk 2]
```diff
@@ -102,45 +117,88 @@ pub fn decode(input: []const u8) DecodeSelfError!Self {
...
+            try std.testing.expectEqual(decoded.value(), num);
+        }
     }
 }
 
 test "ok_full_flow_for_1k_random_values" {
     const rand = std.crypto.random;
-    var buffer = [_]u8{0} ** 9;
+    const zeroed_buffer = [9]u8{ 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+    var buffer = [9]u8{ 0, 0, 0, 0, 0, 0, 0, 0, 0 };
```
Comment (tcoratger) [inline src/types/CompacSizeUint.zig:198]:
```suggestion
    const zeroed_buffer = [_]u8{0} ** 9;
    var buffer = [_]u8{0} ** 9;
```

File: src/types/CompacSizeUint.zig (hunk 1 above)
Comment (tdelabro) [reply src/types/CompacSizeUint.zig]: done

File: src/types/CompacSizeUint.zig (hunk 2 above)
Comment (tdelabro) [reply src/types/CompacSizeUint.zig:198]: done

**Pull request merged**